File: Leetcode/217 - Contains Duplicate/solution.ts
// Naive Solution
function containsDuplicate(nums: number[]): boolean {
    for (let i = 0; i < nums.length; i++) {
        for (let j = i + 1; j < nums.length; j++) {
            if (nums[i] === nums[j]) return true;
        }
    }
    return false;
};

// Better Solution
function containsDuplicate(nums: number[]): boolean {
    nums.sort((a, b) => a - b);
    for (let i = 1; i < nums.length; i++) {
        if (nums[i] === nums[i - 1]) return true;
    }
    return false;
};

// Best Solution
function containsDuplicate(nums: number[]): boolean {
    const numSet = new Set<number>();
    for (let num of nums) {
        if (numSet.has(num)) return true;
        numSet.add(num);
    }
    return false;
};

File: Leetcode/217 - Contains Duplicate/solution.py
# Naive Solution
def containsDuplicate(nums: List[int]) -> bool:
    for i in range(len(nums)):
        for j in range(i + 1, len(nums)):
            if nums[i] == nums[j]:
                return True
    return False

# Better Solution
def containsDuplicate(nums: List[int]) -> bool:
    nums.sort()
    for i in range(1, len(nums)):
        if nums[i] == nums[i - 1]:
            return True
    return False

# Best Solution
def containsDuplicate(nums: List[int]) -> bool:
    num_set = set()
    for num in nums:
        if num in num_set:
            return True
        num_set.add(num)
    return False